### PR TITLE
core: Add option to clone name hints

### PR DESCRIPTION
**Pull request by superlopuh**
The default for this setting is True

### CODE REVIEW

**Review by PapyChacal — APPROVED** [review]
Approving either way, but would be in favour of defaulting to True!

**Review by math-fehr — APPROVED** [review]

**Review by PapyChacal — APPROVED** [review]

**Review by AntonLydike — APPROVED** [review]

File: tests/filecheck/transforms/stencil-unroll.mlir
Comment (webmiche) [inline tests/filecheck/transforms/stencil-unroll.mlir:1]: Did I every mention that I hate SSAValues in the CHECK lines? :sweat_smile:

**Review by webmiche — APPROVED** [review]
LGTM :+1: 

I btw saw you sneaking that `strict=True` in there :wink:

**Review by martin-luecke — APPROVED** [review]
LGTM

**Review by georgebisbas — APPROVED** [review]
Looks good!

**Pull request merged**